### PR TITLE
[3.0] Service Name Mapping Refactor And Nacos model some support.

### PR DESCRIPTION
**Pull request by horizonzy**
Major changes:
1.Nacos model put in order Zookeeper model.
2.Service Name Mapping just support by metadata.
3.Some bug fix.

### CODE REVIEW

**Review by xiaoheng1 — COMMENTED** [review]
getServiceAppMapping and getCasServiceAppMapping move to ServiceNameMapping may be better.

**Review by AlbumenJ — APPROVED** [review]
LGTM.

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java [hunk 1]
```diff
@@ -17,62 +17,109 @@
...
+        execute(() -> {
+            String serviceInterface = url.getServiceInterface();
+            if (IGNORED_SERVICE_INTERFACES.contains(serviceInterface)) {
+                return;
+            }
+            String registryCluster = getRegistryCluster(url);
+            MetadataReport metadataReport = MetadataReportInstance.getMetadataReport(registryCluster);
+            if (metadataReport.isSupportCas() && ServiceNameMappingHandler.isBothMapping()) {
+                doCasMap(metadataReport, url);
+                doNormalMap(metadataReport, url);
```
Comment (chickenlj) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java:60]: Why do we need two different types of mapping method?

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java (hunk 1 above)
Comment (chickenlj) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataServiceNameMapping.java:60]: This is quite confusing on a concept or interface level. Will one single mapping method for all extensions to implement based on their own characteristics work?

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**